diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +131,210 @@ class LineLength : Rule("line-length") {
+        if (node.findChildByType(par)!!.treePrev != null && node.findChildByType(par)!!.treePrev.elementType == WHITE_SPACE)
+            text += node.findChildByType(par)!!.treePrev.text
+        text += node.findChildByType(par)!!.text
+        return text
+    }
+
+    private fun getTextFromParenthesized(node: ASTNode): String {
+        var text = ""
+        text += node.findChildByType(LPAR)!!.text
+        if (node.findChildByType(LPAR)!!.treeNext.elementType == WHITE_SPACE)
+            text += node.findChildByType(LPAR)!!.treeNext.text
+        if (node.findChildByType(RPAR)!!.treePrev.elementType == WHITE_SPACE)
+            text += node.findChildByType(RPAR)!!.treePrev.text
+        text += node.findChildByType(RPAR)!!.text
+        return text
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        if (node.hasChildOfType(BINARY_EXPRESSION) || node.hasChildOfType(PARENTHESIZED) || node.hasChildOfType(POSTFIX_EXPRESSION)) {
+            node.getChildren(null)
+                    .filter { it.elementType == BINARY_EXPRESSION || it.elementType == PARENTHESIZED || it.elementType == POSTFIX_EXPRESSION }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration) {
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            val newParent = parent.findChildByType(PARENTHESIZED) ?: parent
+            if (newParent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
+                fixLongBinaryExpression(newParent, configuration, leftOffset, true)
+            }
+        } else {
+            createNodes(parent, configuration)
+        }
+    }
+
+    private fun createNodes(node: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(node.findChildByType(STRING_TEMPLATE)!!.startOffset).second
+        if (leftOffset > configuration.lineLength)
```

Comment:
```suggestion
        if (leftOffset > configuration.lineLength - STRING_PART_OFFSET)
```
You should probably check this value to avoid errors later in `sbustring` call.